mobile/src/main/java/org/openhab/habdroid/ui/activity/ContentController.java

```diff
@@ -646,16 +673,74 @@ public View onCreateView(@NonNull LayoutInflater inflater,
                 watermark.setVisibility(View.GONE);
             }
 
-            final Button button = view.findViewById(R.id.button);
-            int buttonTextResId = arguments.getInt("buttontext");
+            initButton(arguments, view, R.id.button1, KEY_BUTTON_1_TEXT, KEY_BUTTON_1_TAG);
+            initButton(arguments, view, R.id.button2, KEY_BUTTON_2_TEXT, KEY_BUTTON_2_TAG);
+
+            return view;
+        }
+
+        @Override
+        public void onClick(View view) {
+            if (view.getTag().equals(BUTTON_TAG_OPEN_SETTINGS)) {
+                Intent preferencesIntent = new Intent(getActivity(), PreferencesActivity.class);
+                TaskStackBuilder.create(getActivity())
+                        .addNextIntentWithParentStack(preferencesIntent)
+                        .startActivities();
+            } else if (view.getTag().equals(BUTTON_TAG_ENABLE_DEMO_MODE)) {
+                PreferenceManager.getDefaultSharedPreferences(getContext())
+                        .edit()
+                        .putBoolean(Constants.PREFERENCE_DEMOMODE, true)
+                        .apply();
+            } else if (view.getTag().equals(BUTTON_TAG_ENABLE_WIFI)) {
+                WifiManager wifiManager = (WifiManager)
+                        getContext().getApplicationContext().getSystemService(Context.WIFI_SERVICE);
+                wifiManager.setWifiEnabled(true);
+                ((MainActivity) getActivity()).mController.updateConnection(null,
```

Comment:
`((MainActivity) getActivity()).mController` equals `this`, so you can omit it and just call `updateConnection`

Reply:
Without refactoring `this` is the onClickListener.

Reply:
Oops, right. I thought I had attempted that refactoring, but it looks like I failed at it.
I'll have a look at how to refactor this, I really don't like making `mController` public.